src/js/components/Form/__tests__/Form-test-uncontrolled.js

```diff
@@ -302,75 +392,212 @@ describe('Form uncontrolled', () => {
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('uncontrolled reset without value', () => {
+    const onChange = jest.fn();
+    const { getByPlaceholderText, getByText, queryByText } = render(
+      <Grommet>
+        <Form onChange={onChange}>
+          <FormField
+            name="test"
+            required
+            placeholder="test input"
+            a11yTitle="test"
+          />
+          <Button type="reset" primary label="Reset" />
+        </Form>
+      </Grommet>,
+    );
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: 'Input has changed' },
+    });
+    expect(onChange).toBeCalledTimes(1);
+    fireEvent.click(getByText('Reset'));
+    expect(queryByText('Input has changed')).toBeNull();
+  });
+
+  test('disabled', () => {
+    const onSubmit = jest.fn();
+    const { getByPlaceholderText, getByText, container } = render(
+      <Grommet>
+        <Form onSubmit={onSubmit}>
+          <FormField disabled>
+            <TextInput a11yTitle="test" placeholder="test input" />
+          </FormField>
+          <Button type="submit" primary label="Submit" />
+        </Form>
+      </Grommet>,
+    );
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: 'v' },
+    });
+    fireEvent.click(getByText('Submit'));
+    expect(onSubmit).not.toBeCalledWith(
+      expect.objectContaining({
+        value: { test: 'v' },
+        touched: { test: true },
+      }),
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('regexp validation with status', () => {
+    const onSubmit = jest.fn();
+    const { getByPlaceholderText, getByText, getAllByText } = render(
+      <Grommet>
+        <Form onSubmit={onSubmit}>
+          <FormField
+            name="test"
+            required
+            error="invalid"
+            validate={{
+              regexp: /^[a-z]/i,
+              status: 'info',
+            }}
+            placeholder="test input"
+            a11yTitle="test"
+          />
+          <Button type="submit" primary label="Submit" />
+        </Form>
+      </Grommet>,
+    );
+
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: '1' },
+    });
+    fireEvent.click(getByText('Submit'));
+    expect(getAllByText('invalid')).toMatchSnapshot();
+  });
+
+  test('custom component', () => {
+    const TextIn = ({ name, value, onChange }) => {
+      return (
+        <div>
+          <input
+            type="text"
+            placeholder="Username"
+            name={name}
+            id="test"
+            value={value}
+            onChange={onChange}
+          />
+        </div>
+      );
+    };
+    const onChange = jest.fn();
+    const { getByPlaceholderText } = render(
+      <Grommet>
+        <Form onChange={onChange}>
+          <FormField required placeholder="test input">
+            <TextIn value="Hello World" name="test" onChange={onChange} />
```

Comment:
```suggestion
            <CustomTextInput value="Hello World" name="test" onChange={onChange} />
```

Reply:
Changed name from 'TextIn' to 'CustomTextInput'